corehq/pillows/app_submission_tracker.py

```diff
@@ -36,21 +37,28 @@ def get_app_form_submission_tracker_pillow(pillow_id='AppFormSubmissionTrackerPi
 
 
 class AppFormSubmissionReindexDocProcessor(BaseDocProcessor):
-    def __init__(self, pillow_processor):
+    def __init__(self, pillow_processor, data_source_type, data_source_name):
         self.pillow_processor = pillow_processor
+        self.data_source_type = data_source_type
+        self.data_source_name = data_source_name
 
     def process_doc(self, doc):
-        change = self._doc_to_change(doc)
-        self.pillow_processor.process_change(None, change)
+        change = self._doc_to_change(doc, self.data_source_type, self.data_source_name)
+        try:
+            self.pillow_processor.process_change(None, change)
```

Comment:
I started looking this but I think it's best to keep exception handling here. Currently if an exception gets raised then the interation is aborted which seems like the right thing to do rather than blindly catching all exceptions and assuming it means we should retry the doc.